packages/grafana-ui/src/components/Logs/LogDetails.tsx

```diff
@@ -102,72 +125,92 @@ class UnThemedLogDetails extends PureComponent<Props> {
   };
 
   render() {
-    const { row, theme, onClickFilterOutLabel, onClickFilterLabel, getRows } = this.props;
+    const {
+      row,
+      theme,
+      onClickFilterOutLabel,
+      onClickFilterLabel,
+      getRows,
+      showDuplicates,
+      hasHoverBackground,
+      onChangeHoverBackground,
+    } = this.props;
     const style = getLogRowStyles(theme, row.logLevel);
+    const styles = getStyles(theme);
     const labels = row.labels ? row.labels : {};
     const labelsAvailable = Object.keys(labels).length > 0;
     const fields = this.getAllFields(row);
     const parsedFieldsAvailable = fields && fields.length > 0;
 
     return (
-      <div className={style.logDetailsContainer}>
-        <table className={style.logDetailsTable}>
-          <tbody>
-            {labelsAvailable && (
-              <tr>
-                <td colSpan={5} className={style.logDetailsHeading} aria-label="Log Labels">
-                  Log Labels:
-                </td>
-              </tr>
-            )}
-            {Object.keys(labels).map(key => {
-              const value = labels[key];
-              return (
-                <LogDetailsRow
-                  key={`${key}=${value}`}
-                  parsedKey={key}
-                  parsedValue={value}
-                  isLabel={true}
-                  getStats={() => calculateLogsLabelStats(getRows(), key)}
-                  onClickFilterOutLabel={onClickFilterOutLabel}
-                  onClickFilterLabel={onClickFilterLabel}
-                />
-              );
-            })}
+      <tr
+        className={cx({ [styles.hoverBackground]: hasHoverBackground })}
+        onMouseEnter={() => onChangeHoverBackground(true)}
+        onMouseLeave={() => onChangeHoverBackground(false)}
```

Comment:
From the perspective of making this component loosely coupled to its parent it would make more sense to have more generic naming here. We can just have props:
```
className?: string;
onMouseEnter?: () => void;
onMouseLeave?: () => void;
```

That way it does not need to know about the hover state and would just allow its parent to control it little more. That way the information about the explicit hover logic won't leave the parent and even if that change or be completely removed, this component props would still make sense and wont need to be changed.

Reply:
Yes! That totally makes sense, great great point!